---
title: "2306XX_Sugiphyllo_ITS"
author: "Satoyoshi ISHIZAKI"
date: "2024-01-20"
output: html_document
---

#load packages

```{r}
library(readxl)
library(tidyverse)
library(qiime2R)
library(phyloseq)

library(vegan)
library(zCompositions)

library(GGally)
library(RColorBrewer)
library(ggfortify)
library(ggh4x)

library(dendextend)
library(fantaxtic)
library(ggnested)

library(biomformat)
```

# ASV richness function

```{r, cache=TRUE}
sprich <- function(x) {
  rich <- sum(x > 0)
  return(rich)
}
```

#seed

```{r, cache=TRUE}
seed <- sample(1:1000) #used to set seed
```

#2018_montassier et al_CLOUD-outlier

```{r, cache=TRUE}
# inputs a distance matrix
# returns piecewise distances of samples and their outlier percentile
# and a matrix of the repeated measures of distances  
# k is number of neighbors chosen
"piecewise_kn_V1" <- function(d, test.ix, k=X, ndim=-1){ 
 if(class(d) != 'matrix') d <- as.matrix(d) 
  stats <- numeric(length(test.ix))
  pvals <- numeric(length(test.ix))  
  for(i in 1:length(test.ix)){
    ref.ix <- test.ix[-i]
    keep.ix <- c(test.ix[i], ref.ix)
    if(ndim > -1){
      pc <- cmdscale(d[keep.ix,keep.ix,drop=F],k=ndim)
      d.i <- as.matrix(dist(pc))
    } else {
      d.i <- d[keep.ix,keep.ix,drop=F]
    }
    test.dist <- mean(sort(d.i[1,-1])[1:k])
    ref.dists <- numeric(length(ref.ix))
    for(j in 1:length(ref.ix)){
      ref.dists[j] <- mean(sort(d.i[-1,-1][j,-j]))
    }
    
    stats[i] <- test.dist / mean(ref.dists)
    pvals[i] <- mean(test.dist < ref.dists)
  }
  result <- list()
  result$stats <- stats
  result$pvals <- pvals
  outcome <- pvals <= 0.05
  result$length <- length(outcome[outcome==TRUE]) 
  return(result) 
}
```

# file path

```{r}
path <- "G:/マイドライブ/ishizaki/231202_Sugi_phyllosphere_June"
```

# data import

```{r import, cache.extra=file.mtime("G:/マイドライブ/ishizaki/231202_Sugi_phyllosphere_June/Analysis/NoContam/table-no-contam_ITS.qza"), cache.extra=file.mtime("G:/マイドライブ/ishizaki/231202_Sugi_phyllosphere_June/Analysis/taxa/classification_ITS_single_all.qza"), cache.extra=file.mtime("G:/マイドライブ/ishizaki/231202_Sugi_phyllosphere_June/Analysis/phylogeny/seqs_nocontam_ITS_aligned_masked_tree_rooted.qza"), cache.extra=file.mtime("G:/マイドライブ/ishizaki/231202_Sugi_phyllosphere_June/Analysis/phylogeny/seqs_rarefied_ITS_aligned_masked_tree_rooted.qza")}
otus_ITS <- read_qza(paste0(path, 
                        "/Analysis/NoContam/table-no-contam_ITS.qza"))
taxonomy_ITS <- read_qza(paste0(path, 
                            "/Analysis/taxa/classification_ITS_single_all.qza"))
tree_ITS <- read_qza(paste0(path, "/Analysis/phylogeny/seqs_nocontam_ITS_aligned_masked_tree_rooted.qza"))
tree_rarefied <- read_qza(paste0(path, "/Analysis/phylogeny/seqs_rarefied_ITS_aligned_masked_tree_rooted.qza"))

#load previous data
#load(".Rdata")
```

```{r}
otus_data <- t(otus_ITS$data)
```

#make tax_table

```{r tax_table, cache=TRUE, dependson="import"}
feature_list <- strsplit(as.character(taxonomy_ITS$data$Taxon), ";")
rankmax <- max(sapply(feature_list, length))

# define add_NA(function)
add_NA <- function(vec) {
  length_diff <- rankmax - length(vec)  #difference of length
  vec <- c(vec, rep(NA, length_diff)) #add "NA"
  return(vec)
}

# fill lacking taxonomic ranks with NA
feature_list_fillna <- lapply(feature_list, add_NA)

# tax_table
tax_table <- do.call(rbind, feature_list_fillna)
colnames(tax_table) <- c("Kingdom", 
                        "Phylum", 
                        "Class", 
                        "Order", 
                        "Family", 
                        "Genus", 
                        "Species", 
                        "SpeciesHypothesis")
rownames(tax_table) <- taxonomy_ITS$data$Feature.ID

# ASV_nameとfearture idの対応付け
ASV_name <- paste0(rep("ASV", length(rownames(tax_table))), 
                   seq(1, length(rownames(tax_table))))
tax_table <- cbind(tax_table, ASV_name)
```

# feature numberからASVnameへ

```{r, cache=TRUE, dependson="tax_table"}
feature_row <- sapply(as.list(colnames(otus_data)), 
                      grep, 
                      rownames(tax_table))

otus_data2 <- otus_data
colnames(otus_data2) <- tax_table[feature_row, 9]
```

##tree_feature_name_to_ASV

```{r tree, cache=TRUE, dependson="import", dependson="tax_table"}
tree_ITS2 <- tree_ITS
feature_tree <- sapply(tree_ITS2$data$tip.label, 
                       grep, 
                       rownames(tax_table))
tree_ITS2$data$tip.label <- tax_table[feature_tree, 9]
```

```{r, cache=TRUE, dependson="import", dependson="tax_table"}
tree_rarefied2 <- tree_rarefied
#tip_labels of the phylogenetic tree are also hashes

#replace hashes to ASV names
tree_rarefied2$data$tip.label <- tax_table[sapply(tree_rarefied2$data$tip.label, grep, rownames(tax_table)), 9]
```

#relative abundance(RA)s
log-ratio transformed otus_data has not been used for analysis yet, so jump to rarefaction curve section
##pseudocount

```{r}
otus_data_RA <- otus_data2 / rowSums(otus_data2) #RA, not rarefied

otus_data_pseudo <- otus_data2
otus_data_pseudo <-otus_data_pseudo + 1 #add pseudocount
otus_data_RApseudo <- otus_data_pseudo / rowSums(otus_data_pseudo) #RA after pseudocounts added
```

#AO-relationship

```{r}
plot(log(apply(otus_data_RA, 2, mean)), 
     apply(otus_data_RA, 2, sprich), 
     xlab = "log(Mean Relative Abundance)", 
     ylab = "Occupancy")
abline(h = 16, col = "red")
#AO-relationships
#AO_relationships might be useful to determine the ASV to use as a denominator for alr-transform
```

# rarefaction curves

```{r, cache=TRUE, dependson="import"}
set.seed(123)
rarecurve(otus_data, 
          step = 1000, 
          label = FALSE)
abline(v = min(rowSums(otus_data)), col = "red") #最小read
```

最少リード数時点で種数は概ね飽和しているため、これに合わせてrarefactionする。

# rarefaction

```{r rarefaction, cache=TRUE, dependson="tax_table"}
set.seed(234)
readmin <- min(rowSums(otus_data2))
otus_rarefied <- rrarefy(otus_data2, 
                         readmin)
df_rarefied <- data.frame(otus_rarefied)
df_rarefied[, 1:6]
```

```{r}
#ASVs remained after rarefaction
rarefied_ASV <- colnames(df_rarefied[, apply(df_rarefied, 2, sum) > 0])
rarefied_feature <- rownames(tax_table[charmatch(rarefied_ASV, tax_table[, 9]), ])
rarefied_feature <- data.frame(featureID = rarefied_feature)
#write_tsv(rarefied_feature, paste0(path, "/Analysis/NoContam/hash_ITS_rarefied.tsv"))
```



257 ASVs with occurrence > 10

# metadata

```{r, cache.extra = file.mtime('C:/Users/stysi/iCloudDrive/Documents/研究室関連/samplesheet_map/samplesheet_total.xlsx')}
metadata <- read_excel("C:/Users/stysi/iCloudDrive/Documents/研究室関連/samplesheet_map/samplesheet_total.xlsx", sheet = "samplesheet_1")
metadata <- data.frame(metadata)
rownames(metadata) <- metadata$Sample_Name
metadata$population <- as.factor(metadata$population)
metadata$site <- as.factor(metadata$site)
metadata$dried <- as.factor(metadata$dried)
metadata$var_snip <- as.factor(metadata$var_snip)
metadata$date <- as.factor(metadata$date)

metadata[1:6, ]
```

#alpha diversity

```{r}
reads <- rowSums(otus_data2)
richness <- apply(df_rarefied, 1, sprich)
shannon <- diversity(df_rarefied, index = "shannon")
simpson <- diversity(df_rarefied, index = "simpson")
```

# barplot of samples_no_contam

```{r barplot}
tax_table2 <- tax_table
tax_table2[, 1] <- str_sub(tax_table[, 1], 4)
tax_table2[tax_table2[, 1] == "ssigned", 1] <- tax_table[tax_table[, 1] == "Unassigned", 1]
tax_table2[, 2:8] <- str_sub(tax_table2[, 2:8], 4)

# change feature_name to asv_name.
rownames(tax_table2) <- tax_table2[, 9]
tax_table2 <- tax_table2[, 1:8]

# add clone_ID to rownames (why needed?)
otus_data3 <- otus_data2
rownames(otus_data3) <- paste0(rownames(metadata), "_", metadata$population)

metadata2 <- metadata
metadata2$site <- as.character(metadata2$site)
metadata2$site[metadata2$site == "t"] <- "tsukuba"
metadata2$site[metadata2$site == "k"] <- "kawatabi"
metadata2$site <- as.factor(metadata2$site)

rownames(metadata2) <- paste0(rownames(metadata), "_", metadata$population)

ps_0 <- phyloseq(otu_table(otus_data3[1:50, ], taxa_are_rows = F), 
                 tax_table(tax_table2), 
                 phy_tree(tree_ITS2$data), 
                 sample_data(metadata2[1:50, ])) #change row indices to choose samples included in barplots
```

```{r, cache=TRUE, dependson="barplot"}
# top_taxa(上位10Genus)選別
top_asv <- nested_top_taxa(ps_obj = ps_0, 
                    top_tax_level = "Class", 
                    n_top_taxa = 3, 
                    nested_tax_level = "Family", 
                    n_nested_taxa = 2, 
                    grouping = "site")
```

```{r}
p_comp <- plot_nested_bar(ps_obj = top_asv$ps_obj, 
                      top_level = "Class", 
                      nested_level = "Family")
p_comp <- p_comp + facet_wrap(~ site, 
                      scales = "free_x")
p_comp <- p_comp + labs(y = "Relative Abundance")
p_comp <- p_comp + theme(plot.title = element_text(size = 20),
                         axis.title.x = element_blank(), 
                         #axis.text.x = element_blank(), 
                         axis.title.y = element_text(size = 15), 
                         strip.background.x = element_rect(color = "black", fill = "white"), 
                         strip.text.x = element_text(color =  "black", size = 15))
show(p_comp)
#ggsave(plot = p_comp, filename = "figures/barplot_tsukuba.png")
```

#metadata3

```{r}
#metadata3
metadata3 <- metadata2
rownames(metadata3) <- as.character(metadata3[, 1])
metadata3[, 2] <- gsub("\\.", "_", metadata3[, 2])

metadata3 <- data.frame(metadata3, cloneID = str_sub(metadata3[, 2], 1, 6))
metadata3 <- metadata3[, c(2, 8, 3, 4, 5, 6, 7)]
metadata3$cloneID <- as.factor(metadata3$cloneID)
```

#phyloseq_object

```{r}
ps <- phyloseq(otu_table(otus_data2, taxa_are_rows = FALSE), 
               tax_table(tax_table2), 
               phy_tree(tree_ITS2$data), 
               sample_data(metadata3))
```

#biomformat
to use SCNIC in conda environment, export data as biom files

```{r, eval=FALSE}
col_otudata <- tax_table[feature_row, 9]
attr(col_otudata, "names") <- NULL
otus_biom <- t(otus_data2)
rownames(otus_biom) <- col_otudata
otus_biom <- otu_table(otus_biom, taxa_are_rows = TRUE)

df_biom <- df_rarefied
df_biom <- df_biom[, apply(df_biom, 2, sprich) > 10] #OCC > 10

hash_df <- rownames(tax_table[as.numeric(str_sub(colnames(df_biom), 4, 7)), ]) #hashes for each columns of df_rarefied
hash_df <- data.frame(featureid = hash_df)

colnames(df_biom) <- hash_df$featureid
df_biom <- make_biom(t(df_biom))


tax_biom <- data.frame(tax_table2[feature_row, ])
tax_biom <- rownames_to_column(tax_biom)


biom_obj <- make_biom(data = otus_biom)
```

```{r}
#write_biom(biom_obj, biom_file = paste0(path, "/Analysis/biom/biom_ITS.biom"))
#write_tsv(tax_biom, file = paste0(path, "/Analysis/network/network_ITS/tax_ITS.tsv"))
#write_biom(df_biom, biom_file = paste0(path, "/Analysis/picrust2_in/df_ITS.biom"))
#write_tsv(hash_df, file = paste0(path, "/Analysis/picrust2_in/hash_df_ITS.tsv"))
```

# NMDS
## dist, otus_data

```{r dist, cache=TRUE, dependson="rarefaction", dependson="tree"}
otus_rel <- df_rarefied/rowSums(df_rarefied)
otus_exabs <- df_rarefied
otus_exabs[otus_exabs > 0] <- 1

dist_otus <- vegdist(df_rarefied, method = "bray")
dist_jacc <- vegdist(otus_exabs, method = "jaccard")

dist_UF <- UniFrac(phyloseq(otu_table(df_rarefied, taxa_are_rows = FALSE), 
                                phy_tree(tree_rarefied2$data)), 
                       weighted = FALSE)
dist_UF_wei <- UniFrac(phyloseq(otu_table(df_rarefied, taxa_are_rows = FALSE), 
                                phy_tree(tree_rarefied2$data)), 
                       weighted = TRUE)

```

#Outlier

Using CLOUD analysis (developed by E. Montassier et al., 2018) to detect outlier samples

```{r}
outlier_bray <- list(tsukuba = piecewise_kn_V1(dist_otus, 
                                               which(metadata3$site == "tsukuba"), 
                                               k = 5), 
                     kawatabi = piecewise_kn_V1(dist_otus, 
                                                which(metadata3$site == "kawatabi"), 
                                                k = 5))
outlier_UFwei <- list(tsukuba = piecewise_kn_V1(dist_UF_wei, 
                                                which(metadata3$site == "tsukuba"), 
                                                k = 5), 
                      kawatabi = piecewise_kn_V1(dist_UF_wei, 
                                                 which(metadata3$site == "kawatabi"), 
                                                 k = 5))

outlier_bray <- rbind(metadata3[which(outlier_bray$tsukuba$pvals < 0.05), ], 
                      metadata3[50 + which(outlier_bray$kawatabi$pvals < 0.05), ])
outlier_UFwei <- rbind(metadata3[which(outlier_UFwei$tsukuba$pvals < 0.05), ],
                       metadata3[50 + which(outlier_UFwei$kawatabi$pvals < 0.05), ])
show(outlier_bray)
show(outlier_UFwei)
```

Should I exlude these samples?(especially when ordination?)
##bray

```{r, cache=TRUE, dependson="dist"}
stress_value <- numeric(100)
for (i in 1:100) {
  set.seed(i)
  nmds_otus <- metaMDS(dist_otus, trymax = 100)
  stress_value[i] <- nmds_otus$stress
}
```

```{r, cache=TRUE, dependson="dist"}
set.seed(grep(min(stress_value), stress_value))
nmds_otus <- metaMDS(dist_otus, trymax = 100)
```

```{r}
stressplot(nmds_otus)
nmds_otus$stress
```

```{r, fig.asp=0.7}
df_nmds <- data.frame(metadata3, nmds_otus$points)

replace_rule <- c("AJ"=1, "NB"=2, "YM"=3, "IS"=4, "DN"=5, "BJ"=6, "AS"=7, "OK"=8, "TY"=9, "AZ"=10, "SM"=11, "KW"=12, "SN"=13, "YK"=14) #概ね緯度に沿って番号割り振り
pop_num <- as.character(df_nmds$population)
pop_num <- unlist(lapply(pop_num, function(x) replace_rule[x]))
fac_sipop <- as.factor(paste0(df_nmds$site, LETTERS[pop_num])) #pop, siteでfactor化
col_sipop <- c(alpha("orange3", seq(0.5, 1, length.out = 14)), 
               alpha("blue", seq(0.5, 1, length.out = 14)))


p_nmds <- ggplot(data = df_nmds, 
                 mapping = aes(x = MDS1, y = MDS2, color = fac_sipop))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + scale_color_manual(values = col_sipop)
p_nmds <- p_nmds + geom_text(aes(label = LETTERS[pop_num]))
p_nmds <- p_nmds + annotate("text", 
                            x = 1.3, 
                            y = -1, 
                            label = paste0("stress = ", 
                                           round(nmds_otus$stress, digits = 4)))
p_nmds <- p_nmds + theme(legend.position = "none")
p_nmds
#ggsave(filename = "figures/NMDS_bray_sipop.png", plot = p_nmds)
```

```{r}
# population, cloneIDでまとめる
df_nmds_out <- cbind(df_nmds, outlier = rep(NA, nrow(df_nmds)))
df_nmds_out[rownames(outlier_bray), "outlier"] <- rep("Out", nrow(outlier_bray))

p_nmds <- ggplot(data = df_nmds_out, 
                 mapping = aes(x = MDS1, y = MDS2))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + geom_point(aes(color = site))
p_nmds <- p_nmds + geom_text(aes(label = outlier))
p_nmds <- p_nmds + annotate("text", 
                            x = 1.2,y = -2, 
                            label = paste0("stress = ", 
                                           round(nmds_otus$stress, digits = 4)))
p_nmds
#rm(df_nmds_out)
```

##weighted UniFrac

```{r}
#exclude outliers
df_rarefied_out <- df_rarefied[-charmatch(rownames(outlier_UFwei), 
                                          rownames(df_rarefied)
                                          ), ]
metadata_out <- metadata3[-charmatch(rownames(outlier_UFwei), 
                                          rownames(df_rarefied)
                                          ), ]

dist_UF_wei_out <- UniFrac(phyloseq(otu_table(df_rarefied_out, taxa_are_rows = FALSE), 
                                    phy_tree(tree_rarefied2$data)), 
                           weighted = TRUE)
```

```{r, echo=FALSE, cache=TRUE, dependson="dist"}
stress_UF_wei <- numeric(100)
for (i in 1:100) {
  set.seed(i)
  nmds_UF_wei <- metaMDS(dist_UF_wei_out, trymax = 100)
  stress_UF_wei[i] <- nmds_UF_wei$stress
}
```

```{r, cache=TRUE, dependson="dist"}
set.seed(grep(min(stress_UF_wei), stress_UF_wei))
nmds_UF_wei <- metaMDS(dist_UF_wei_out, trymax = 100)
```

```{r}
stressplot(nmds_UF_wei)
```

```{r, eval = FALSE}
#the distortion of the plot is not due to the ouliers ?
df_UF_wei <- data.frame(metadata_out, nmds_UF_wei$points)

p_nmds <- ggplot(data = df_UF_wei, 
                 mapping = aes(x = MDS1, y = MDS2, 
                               color = fac_sipop[-charmatch(rownames(outlier_UFwei), 
                                                            rownames(df_rarefied))]
                               )
                 )
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + scale_color_manual(values = col_sipop)
p_nmds <- p_nmds + geom_text(aes(label = LETTERS[pop_num[-charmatch(rownames(outlier_UFwei), 
                                                                    rownames(df_rarefied))]]))
p_nmds <- p_nmds + annotate("text", 
                            x = 0.02, y = -0.02, 
                            label = paste0("stress = ", 
                                           round(nmds_UF_wei$stress, digits = 4)))
p_nmds <- p_nmds + theme(legend.position = "none")
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_weightedUF_exclude.png")
```

```{r}
df_UF_wei <- data.frame(metadata3, nmds_UF_wei$points)

p_nmds <- ggplot(data = df_UF_wei, 
                 mapping = aes(x = MDS1, y = MDS2,color = fac_sipop))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + scale_color_manual(values = col_sipop)
p_nmds <- p_nmds + geom_text(aes(label = LETTERS[pop_num]))
p_nmds <- p_nmds + annotate("text", 
                            x = 0.017, y = -0.015, 
                            label = paste0("stress = ", 
                                           round(nmds_UF_wei$stress, digits = 4)))
p_nmds <- p_nmds + theme(legend.position = "none")
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_weightedUF.png")
```

#ASV fitting
##filter Class
```{r, cache=TRUE, dependson="rarefaction"}
ps_rarefied <- phyloseq(otu_table(df_rarefied, taxa_are_rows = FALSE),
                        tax_table(tax_table2),
                        phy_tree(tree_ITS2$data),
                        sample_data(metadata3))

#merge ASVs at Class level
ps_rarefied_Class <- tax_glom(ps_rarefied, taxrank = "Class")
```

```{r}
#fit ASV abundance to NMDS plots
plot(log(apply(otu_table(ps_rarefied_Class) / 
          rowSums(otu_table(ps_rarefied_Class)), 
               2, mean)), 
     apply(otu_table(ps_rarefied_Class), 2, sprich), 
     xlab = "log(Mean Relative Abundance)", 
     ylab = "Occupancy")
abline(h = 30, col = "red")
```

```{r}
#filter Class with <= 30 occurrences
Abundant_Class <- which(apply(otu_table(ps_rarefied_Class), 2, sprich) > 30)
df_Class_abundant <- otu_table(ps_rarefied_Class)[, Abundant_Class]
```

##filter ASV

```{r}
plot(log(apply(otus_data_RA, 2, mean)), 
     apply(otus_data_RA, 2, sprich), 
     xlab = "log(Mean Relative Abundance)", 
     ylab = "Occupancy")
abline(v = -6.2, col = "red")
abline(h = 20, col = "red")
```

```{r}
#filter Genus with <= 30 occurrences
Abundant_asv <- which((log(apply(otus_data_RA, 2, mean)) > -6.2) &
                        (apply(otus_data_RA, 2, sprich) > 20))
df_asv_abundant_out <- df_rarefied_out[, Abundant_asv]
```

##fit Class

```{r}
set.seed(seed[1])
fit_Class_bray <- envfit(nmds_otus, df_Class_abundant)
show(fit_Class_bray)
```

```{r}
score <- scores(fit_Class_bray, display = "vectors")
score <- rownames_to_column(data.frame(score))
score <- data.frame(score, 
                    Class = tax_table(ps_rarefied_Class)[score$rowname, "Class"], 
                    p = fit_Class_bray$vectors$pvals,
                    arr_col = rep("grey40", nrow(score)))
score$arr_col[score$p < 0.05] <- rep("black", sum(score$p < 0.05))
```

```{r}
p_nmds <- ggplot(data = df_nmds, 
                 mapping = aes(x = MDS1, y = MDS2, color = site))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + geom_point(aes())
p_nmds <- p_nmds + geom_text(data = score,
                             aes(x = NMDS1*3, y = NMDS2*3, 
                                 label = Class),
                             size = 3, 
                             color = score$arr_col) 
p_nmds <- p_nmds + annotate("text", 
                            x = 1.0,y = -2, 
                            label = paste0("stress = ", 
                                           round(nmds_otus$stress, 
                                                 digits = 4)))
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_fitclass.png")
```

##fit asv bray

```{r}
df_asv_abundant <- df_rarefied[, Abundant_asv]
set.seed(seed[9])
fit_asv_bray <- envfit(nmds_otus, df_asv_abundant)
show(fit_asv_bray)
```

```{r}
score <- scores(fit_asv_bray, display = "vectors")
score <- rownames_to_column(data.frame(score))
score <- data.frame(score, 
                    asv = tax_table2[score$rowname, "Family"], 
                    p = fit_asv_bray$vectors$pvals,
                    arr_col = rep("grey40", nrow(score)))
score$arr_col[score$p < 0.05] <- rep("black", sum(score$p < 0.05))
```

```{r}
p_nmds <- ggplot(data = df_nmds, 
                 mapping = aes(x = MDS1, y = MDS2, color = site))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + geom_point(aes())
p_nmds <- p_nmds + geom_text(data = score[score$p < 0.05, ],
                             aes(x = NMDS1*3, y = NMDS2*3, label = asv),
                             size = 3, 
                             color = score[score$p < 0.05, ]$arr_col)
p_nmds <- p_nmds + annotate("text", 
                            x = 1.0,y = -2, 
                            label = paste0("stress = ", 
                                           round(nmds_otus$stress, 
                                                 digits = 4)))
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_fitclass.png")
```

##fit Class weighted UF

```{r}
set.seed(seed[2])
fit_Class_UFwei <- envfit(nmds_UF_wei, df_Class_abundant)
show(fit_Class_UFwei)
```

```{r}
score_UFwei <- scores(fit_Class_UFwei, display = "vectors")
score_UFwei <- rownames_to_column(data.frame(score_UFwei))
score_UFwei <- data.frame(score_UFwei, 
                    Class = tax_table(ps_rarefied_Class)[score_UFwei$rowname, "Class"], 
                    p = fit_Class_UFwei$vectors$pvals,
                    arr_col = rep("grey", nrow(score_UFwei)))
score_UFwei$arr_col[score_UFwei$p < 0.05] <- rep("black", sum(score_UFwei$p < 0.05))
```

```{r}
p_nmds <- ggplot(data = df_UF_wei, 
                 mapping = aes(x = MDS1, y = MDS2, color = site))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + geom_point(aes())
p_nmds <- p_nmds + geom_text(data = score_UFwei, 
                             aes(x = NMDS1/10, y = NMDS2/10, 
                                 label = Class),
                             size = 3, 
                             color = score_UFwei$arr_col) 
p_nmds <- p_nmds + annotate("text", 
                            x = 0.017, y = -0.02, 
                            label = paste0("stress = ", 
                                           round(nmds_UF_wei$stress, digits = 4)))
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_fitclass_weightedUF.png")
```

##fit asv weighted UF

```{r}
set.seed(seed[4])
fit_asv_UFwei <- envfit(nmds_UF_wei, df_asv_abundant_out)
show(fit_asv_UFwei)
```

```{r}
score_UFwei <- scores(fit_asv_UFwei, display = "vectors")
score_UFwei <- rownames_to_column(data.frame(score_UFwei))
score_UFwei <- data.frame(score_UFwei, 
                    asv = tax_table2[score_UFwei$rowname, "Family"], 
                    p = fit_asv_UFwei$vectors$pvals,
                    arr_col = rep("grey", nrow(score_UFwei)))
score_UFwei$arr_col[score_UFwei$p < 0.05] <- rep("black", sum(score_UFwei$p < 0.05))
```

```{r}
p_nmds <- ggplot(data = df_UF_wei, 
                 mapping = aes(x = MDS1, y = MDS2, color = site))
p_nmds <- p_nmds + theme_bw()
p_nmds <- p_nmds + coord_fixed(1)
p_nmds <- p_nmds + xlim(-0.07, 0.04)
p_nmds <- p_nmds + geom_point(aes())
p_nmds <- p_nmds + geom_text(data = score_UFwei, 
                             aes(x = NMDS1/10, y = NMDS2/10, 
                                 label = asv),
                             size = 3, 
                             color = score_UFwei$arr_col) 
p_nmds <- p_nmds + annotate("text", 
                            x = 0.023, y = -0.04, 
                            label = paste0("stress = ", 
                                           round(nmds_UF_wei$stress, digits = 4)), 
                            size = 3.5)
show(p_nmds)
#ggsave(plot = p_nmds, file = "figures/NMDS_fitASV_weightedUF.png")
```

# PERMANOVA
##bray

```{r}
set.seed(23)
model <- adonis2(dist_otus ~ site + population, 
                 data = metadata3, 
                 perm = 999, 
                 method = "bray")
model
```

```{r}
set.seed(345)
model <- adonis2(dist_otus ~ site * population, 
                 data = metadata3, 
                 perm = 999, 
                 method = "bray")
model
```

```{r}
set.seed(567)
adonis2(vegdist(df_rarefied[metadata3$site == "tsukuba", ]) ~ population, 
        data = metadata3[metadata3$site == "tsukuba", ], 
        perm = 999, 
        method = "bray")

set.seed(678)
adonis2(vegdist(df_rarefied[metadata3$site == "kawatabi", ]) ~ population, 
        data = metadata3[metadata3$site == "kawatabi", ], 
        perm = 999, 
        method = "bray")
```

##UniFrac

```{r}
set.seed(seed[3])
adonis2(dist_UF_wei ~ site * population, 
                 data = metadata3, 
                 perm = 999, 
                 method = "bray")
```

# PERMDISP (to test homogenity of dispersion)

```{r}
set.seed(456)
dispersion <- betadisper(dist_otus, 
                         metadata$site, 
                         type = "centroid")
dispersion
permutest(dispersion)
plot(dispersion)
```

```{r}
#permdisp tests for homogeneity of variance
set.seed(seed[5])
dispersion_UF_wei <- betadisper(dist_UF_wei_out, 
                         metadata_out$site, 
                         type = "centroid")
dispersion_UF_wei
permutest(dispersion_UF_wei)

plot(dispersion_UF_wei)
```

# comparison of sp richness and alpha diversity

```{r, fig.asp = 0.7}
p_rich <- ggplot(data = metadata2, 
                 mapping = aes(x = site, y = richness)) + 
  geom_boxplot(aes(colour = site)) + 
  labs(title = "ASV richness")

p_shannon <- ggplot(data = metadata2, 
                    mapping = aes(x = site, y = shannon)) + 
  geom_boxplot(aes(colour = site)) + 
  labs(title = "Shannon indices")

p_simpson <- ggplot(data = metadata2, 
                    mapping = aes(x = site, y = simpson)) + 
  geom_boxplot(aes(colour = site)) + 
  labs(title = "Simpson indices")

p_rich
p_shannon
p_simpson
```

```{r}
set.seed(567)
kruskal.test(richness ~ metadata$site)
kruskal.test(shannon ~ metadata$site)
kruskal.test(simpson ~ metadata$site)
```

#BVOC


```{r, import_BVOC, cache.extra = file.mtime('C:/Users/stysi/iCloudDrive/Documents/研究室関連/samplesheet_map/BVOC/BVOC_summary_240122.xlsx')}
BVOC_kawatabi <- read_excel("/Users/ishizakisatoyoshi/Documents/研究室関連/samplesheet_map/BVOC/BVOC_summary_240122.xlsx", sheet = "2023summer_KAWATABI")
BVOC_kawatabi <- t(BVOC_kawatabi)
colnames(BVOC_kawatabi) <- BVOC_kawatabi[1, ]
BVOC_kawatabi <- BVOC_kawatabi[2:nrow(BVOC_kawatabi), ]

tree_ID <- as.numeric(str_sub(rownames(BVOC_kawatabi), 6, 7))
tree_ID[is.na(tree_ID) == TRUE] <- 1
BVOC_ID <- paste0(str_sub(rownames(BVOC_kawatabi), 1, 2), 
                  "_", 
                  str_sub(rownames(BVOC_kawatabi), 3, 5), 
                  "_", 
                  tree_ID)

BVOC_kawatabi <- as.data.frame(lapply(data.frame(BVOC_kawatabi), as.numeric))
rownames(BVOC_kawatabi) <- BVOC_ID

BVOC <- sapply(strsplit(colnames(BVOC_kawatabi), "\\."), '[', 1)
BVOC[BVOC == "a"] <- "α"
BVOC[BVOC == "b"] <- "β"
BVOC[BVOC == "g"] <- "γ"
BVOC[BVOC == "X3"] <- "3"
BVOC <- paste0(BVOC, "-", sapply(strsplit(colnames(BVOC_kawatabi), "\\."), '[', 2))
BVOC <- sub("-NA", "", BVOC)
colnames(BVOC_kawatabi) <- BVOC
show(BVOC_kawatabi)
```

```{r}
#barplot
BVOC_k_ga <- rownames_to_column(BVOC_kawatabi[, colSums(BVOC_kawatabi) > 0], var = "ID")
BVOC_k_ga <- gather(BVOC_k_ga, "BVOC", "emission", -ID)
col_BVOC <- colorRampPalette(brewer.pal(9, "Set1"))(ncol(BVOC_kawatabi[, colSums(BVOC_kawatabi) > 0]))

p_BVOC <- ggplot(data = BVOC_k_ga, mapping = aes(x = ID, y = emission, 
                                                 fill = as.factor(BVOC)))
p_BVOC <- p_BVOC + geom_bar(stat = "identity")
p_BVOC <- p_BVOC + scale_fill_manual(values = col_BVOC)
p_BVOC <- p_BVOC + theme_bw()
p_BVOC <- p_BVOC + labs(title = "Basal emission rate of BVOCs", 
                        fill = "BVOC")
p_BVOC <- p_BVOC + ylab("Basal emission rate (ng / (gdw h))")
p_BVOC <- p_BVOC + theme(axis.text.x = element_text(size = 6, 
                                                    angle = 90, 
                                                    margin = margin(t = 5)), 
                         legend.key.size = unit(0.2, "cm"))
show(p_BVOC)
```

##process data

```{r process_data, cache=TRUE, dependson="import_BVOC"}
#"AZ_004_1", "AJ_002_1"はphyllosphereの方にないので除く
#"AZ_040_1"はNMDSで外れ値となるので一旦除く

phyllo_VOC <- sapply(rownames(BVOC_kawatabi[-sapply(c("AZ_004_1", "AJ_002_1", "AZ_040_1"), 
                                                    grep, 
                                                    rownames(BVOC_kawatabi)), ]), 
                     grep, 
                     metadata3$tree_ID[metadata3$site == "kawatabi"])

df_rarefied_VOC <- df_rarefied[(50 + phyllo_VOC), ] #VOC測定した個体の葉圏
metadata_VOC <- metadata3[(50 + phyllo_VOC), ]
BVOC_phyllo <- BVOC_kawatabi[-sapply(c("AZ_004_1", "AJ_002_1", "AZ_040_1"), 
                                     grep, 
                                     rownames(BVOC_kawatabi)), ]
BVOC_phyllo <- BVOC_phyllo[, colSums(BVOC_phyllo) > 0]

phyllo_VOC_UFwei <- sapply(rownames(BVOC_kawatabi[-sapply(c("AZ_004_1", "AJ_002_1"), 
                                                    grep, 
                                                    rownames(BVOC_kawatabi)), ]), 
                     grep, 
                     metadata3$tree_ID[metadata3$site == "kawatabi"])

df_rarefied_VOC_UFwei <- df_rarefied[(50 + phyllo_VOC_UFwei), ] #VOC測定した個体の葉圏
metadata_VOC_UFwei <- metadata3[(50 + phyllo_VOC_UFwei), ]
BVOC_phyllo_UFwei <- BVOC_kawatabi[-sapply(c("AZ_004_1", "AJ_002_1"), 
                                     grep, 
                                     rownames(BVOC_kawatabi)), ]
BVOC_phyllo_UFwei <- BVOC_phyllo_UFwei[, colSums(BVOC_phyllo_UFwei) > 0]
df_rarefied_VOC_UFwei2 <- df_rarefied_VOC_UFwei
rownames(df_rarefied_VOC_UFwei2) <- metadata_VOC_UFwei$tree_ID

BVOC_total <- rowSums(BVOC_phyllo)

BVOC_rel <- BVOC_phyllo / rowSums(BVOC_phyllo)
BVOC_clr <- clr(BVOC_rel)
BVOC_pseudo <- BVOC_phyllo + 0.01
BVOC_log <- log(BVOC_pseudo)
BVOC_exabs <- BVOC_phyllo
BVOC_exabs[BVOC_exabs > 0] <- 1

BVOC_totalog <- log(rowSums(BVOC_pseudo))

BVOC_pseudo_UFwei <- BVOC_phyllo_UFwei + 0.01
BVOC_log_UFwei <- log(BVOC_pseudo_UFwei)
BVOC_totalog_UFwei <- log(rowSums(BVOC_pseudo_UFwei))
```

```{r, cache=TRUE, dependson="process_data"}
df_rarefied_VOC2 <- df_rarefied_VOC
rownames(df_rarefied_VOC2) <- metadata_VOC$tree_ID
metadata_VOC2 <- metadata_VOC
rownames(metadata_VOC2) <- metadata_VOC2$tree_ID

ps_VOC <- phyloseq(otu_table(df_rarefied_VOC2, taxa_are_rows = F), 
                   tax_table(tax_table2), 
                   sample_data(metadata_VOC2))

top_asv2 <- nested_top_taxa(ps_obj = ps_VOC, 
                           top_tax_level = "Class", 
                           n_top_taxa = 4, 
                           nested_tax_level = "Genus", 
                           n_nested_taxa = 2)
```

```{r, fig.asp=1}
p_comp_2 <- plot_nested_bar(ps_obj = top_asv2$ps_obj, 
                      top_level = "Class", 
                      nested_level = "Genus")
p_comp_2 <- p_comp_2 + labs(title = "Phyllosphere fungal community composition", 
                        y = "Relative Abundance")
p_comp_2 <- p_comp_2 + facet_wrap(~ population, 
                                  scales = "free_x")
p_comp_2 <- p_comp_2 + theme(plot.title = element_text(size = 15), 
                         axis.title.x = element_blank(), 
                         axis.text.x = element_text(size = 8, 
                                                    angle = 90, 
                                                    margin = margin(t = 5, 
                                                                    b = 10)), 
                         axis.title.y = element_text(size = 10), 
                         strip.background.x = element_rect(color = "black", fill = "white"), 
                         strip.text.x = element_text(color =  "black", size = 10))
show(p_comp_2)
```

##biomfile

```{r, eval=FALSE}
"otus_VOC <- t(df_rarefied_VOC2)
otus_VOC <- otus_VOC[apply(otus_VOC, 1, mean) >= 2, ]
otus_VOC <- otu_table(otus_VOC, taxa_are_rows = TRUE)
colSums(otus_VOC) / rowSums(df_rarefied_VOC2)

tax_row_VOC <- sapply(rownames(otus_VOC), charmatch, rownames(tax_table2))
tax_VOC <- data.frame(tax_table2[tax_row_VOC, ])
tax_VOC <- rownames_to_column(tax_VOC)" #didn't_use

otus_VOC_rare <- t(df_rarefied_VOC2[, 
                                    apply((df_rarefied_VOC2 / rowSums(df_rarefied_VOC2)), 2, mean) >= 0.001]
                   ) #MRA >= 0.001

otus_VOC_rare2 <- t(df_rarefied_VOC_UFwei2[, 
                                    apply(df_rarefied_VOC_UFwei2, 2, sprich) > 10]
                   ) #rarefied ASV table, occurrence > 10
otus_VOC_rare2_bm <- make_biom(data = otus_VOC_rare2)

tax_row_VOC_rare <- sapply(rownames(otus_VOC_rare), charmatch, rownames(tax_table2))
tax_VOC_rare <- data.frame(tax_table2[tax_row_VOC_rare, ])
tax_VOC_rare <- rownames_to_column(tax_VOC_rare)

biom_BVOC <- otu_table(t(BVOC_phyllo), taxa_are_rows = TRUE)

otus_VOC_rare_bm <- make_biom(otus_VOC_rare)
otus_VOC_bm <- make_biom(data = otus_VOC)
BVOC_bm <- make_biom(data = biom_BVOC)
```

```{r}
#write_biom(otus_VOC_bm, biom_file = paste0(path, "/Analysis/biom/otus_BVOC_ITS.biom"))
#write_biom(BVOC_bm, biom_file = paste0(path, "/Analysis/biom/BVOC_ITS.biom"))
#write_biom(otus_VOC_rare_bm, biom_file = paste0(path, "/Analysis/biom/otus_BVOC_ITS_MRA.biom"))
#write_biom(otus_VOC_rare2_bm, biom_file = paste0(path, "/Analysis/biom/otus_BVOC_ITS_rf_OCC.biom"))
#write_tsv(tax_VOC_rare, file = paste0(path, "/Analysis/SCNIC_py/tax_BVOC_ITS_MRA.tsv"))
```

##pca(BVOC)

```{r}
pca_BVOC <- prcomp(BVOC_phyllo, scale. = TRUE)
summary(pca_BVOC)
```

```{r}
biom_BVOC_PCA <- otu_table(t(pca_BVOC$x[, 1:4]), taxa_are_rows = TRUE)
BVOC_PCA_bm <- make_biom(data = biom_BVOC_PCA)
write_biom(BVOC_PCA_bm, biom_file = paste0(path, "/Analysis/biom/BVOC_PCA_ITS.biom"))
```

```{r}
terpene <- c(rep("grey35", 12), "blue3", "purple2", "green3")

p_pca <- autoplot(pca_BVOC, 
                  data = data.frame(BVOC_phyllo, population = metadata_VOC$population), 
                  color = "population", 
                  loadings = TRUE, 
                  loadings.color = "grey85", 
                  loadings.label = TRUE, 
                  loadings.label.color = terpene[-15], 
                  loadings.label.size = 3.5) + 
  coord_fixed(ratio = 1) + 
  theme_bw()
#ggsave(filename = "figures/PCA_12.png", plot = p_pca)
show(p_pca)
```

```{r}
p_pca <- autoplot(pca_BVOC, 
                  data = data.frame(BVOC_phyllo, population = metadata_VOC$population), 
                  x = 3, 
                  y = 4, 
                  color = "population", 
                  loadings = TRUE, 
                  loadings.color = "grey85", 
                  loadings.label = TRUE, 
                  loadings.label.color = terpene[-15], 
                  loadings.label.size = 3.5) + 
  coord_fixed(ratio = 1) + 
  theme_bw() + 
  scale_x_continuous(limits = c(-0.75, 0.75))
#ggsave(filename = "figures/PCA_34.png", plot = p_pca)
show(p_pca)
```

##pca(log(BVOC))

```{r}
#remove kaurene from dataframe
#sapply(c("kaurene"), grep, colnames(BVOC_log_ps))
pca_BVOC_log_rm <- prcomp(data.frame(BVOC_log[, -c(14)], 
                                     total = BVOC_totalog), 
                          scale. = TRUE)
summary(pca_BVOC_log_rm)

p_pca <- autoplot(pca_BVOC_log_rm, 
                  data = data.frame(BVOC_log[, -c(14)], 
                                    total = BVOC_totalog,
                                    population = metadata_VOC$population), 
                  x = 1, 
                  y = 2,
                  color = "population", 
                  loadings = TRUE, 
                  loadings.color = "grey85", 
                  loadings.label = TRUE, 
                  loadings.label.color = terpene[-c(14)]) + 
  coord_fixed(ratio = 1) + 
  theme_bw()
#ggsave(filename = "figures/PCA_log_rm_12.png", plot = p_pca)
show(p_pca)
```

##NMDS(for envfit)

```{r, echo=FALSE, cache=TRUE, dependson="process_data"}
stress_value_BVOC <- numeric(100)
dist_otus_BVOC <- vegdist(df_rarefied_VOC, method = "bray")
for (i in 1:100) {
  set.seed(i)
  nmds_otus_BVOC <- metaMDS(dist_otus_BVOC, trymax = 100)
  stress_value_BVOC[i] <- nmds_otus_BVOC$stress
}

set.seed(grep(min(stress_value_BVOC), stress_value_BVOC))
nmds_otus_BVOC <- metaMDS(dist_otus_BVOC, trymax = 100)
```

```{r}
stressplot(nmds_otus_BVOC)
nmds_otus_BVOC$stress
```

```{r}
df_nmds_BVOC <- data.frame(metadata_VOC, 
                        VOC_total = rowSums(BVOC_phyllo), 
                        nmds_otus_BVOC$points)

p_nmds_BVOC <- ggplot(data = df_nmds_BVOC, 
                 mapping = aes(x = MDS1, y = MDS2))
p_nmds_BVOC <- p_nmds_BVOC + theme_bw()
p_nmds_BVOC <- p_nmds_BVOC + geom_point(aes(color = population))
p_nmds_BVOC <- p_nmds_BVOC + geom_path(aes(group = cloneID))
p_nmds_BVOC <- p_nmds_BVOC + annotate("text", 
                            x = 0.75, 
                            y = -1, 
                            label = paste0("stress = ", 
                                           round(nmds_otus_BVOC$stress, digits = 4)))
p_nmds_BVOC <- p_nmds_BVOC + labs(title = "NMDS plot, kawatabi, VOC sampled")
p_nmds_BVOC
```

```{r}
#permanova
set.seed(seed[6])
adonis2(dist_otus_BVOC ~ population, 
       data = metadata_VOC, 
       perm = 999)
```

##NMDS weightedUF

```{r, cache=TRUE, echo=FALSE, dependson="process_data"}
stress_BVOC_UFwei <- numeric(100)

dist_otus_BVOC_UFwei <- UniFrac(phyloseq(otu_table(t(df_rarefied_VOC_UFwei), 
                                                   taxa_are_rows = TRUE), 
                                         phy_tree(tree_rarefied2$data)), weighted = TRUE)
for (i in 1:100) {
  set.seed(i)
  nmds_otus_BVOC_UFwei <- metaMDS(dist_otus_BVOC_UFwei, trymax = 100)
  stress_BVOC_UFwei[i] <- nmds_otus_BVOC_UFwei$stress
}
```

```{r, echo=FALSE}
set.seed(grep(min(stress_BVOC_UFwei), stress_BVOC_UFwei))
nmds_otus_BVOC_UFwei <- metaMDS(dist_otus_BVOC_UFwei, trymax = 100)
```

```{r}
stressplot(nmds_otus_BVOC_UFwei)
nmds_otus_BVOC_UFwei$stress
```

```{r}
df_nmds_BVOC_UFwei <- data.frame(metadata_VOC_UFwei, 
                                 total = BVOC_totalog_UFwei, 
                                 nmds_otus_BVOC_UFwei$points)

p_nmds_BVOC <- ggplot(data = df_nmds_BVOC_UFwei, 
                 mapping = aes(x = MDS1, y = MDS2))
p_nmds_BVOC <- p_nmds_BVOC + theme_bw()
p_nmds_BVOC <- p_nmds_BVOC + coord_fixed(1)
p_nmds_BVOC <- p_nmds_BVOC + geom_point(aes(color = population))
#p_nmds_BVOC <- p_nmds_BVOC + geom_path(aes(group = cloneID))
p_nmds_BVOC <- p_nmds_BVOC + annotate("text", 
                            x = 0.035, 
                            y = -0.01, 
                            label = paste0("stress = ", 
                                           round(nmds_otus_BVOC_UFwei$stress, digits = 4)))
p_nmds_BVOC
#ggsave(plot = p_nmds_BVOC, file = "figures/NMDS_BVOC_UFwei.png")
```

##envfit

```{r}
set.seed(34)
fit_VOC <- envfit(nmds_otus_BVOC, cbind(pca_BVOC_log_rm$x[, 1:2], BVOC_totalog))
show(fit_VOC)

VOC_score <- scores(fit_VOC, display = "vectors")
VOC_score <- rownames_to_column(data.frame(VOC_score))
VOC_score <- data.frame(VOC_score, 
                        p = fit_VOC$vectors$pvals, 
                        arr_col = rep("grey", nrow(VOC_score)))
```

```{r}
p_fit <- ggplot(df_nmds_BVOC) +
  theme_bw() + 
  geom_point(mapping = aes(x = MDS1, y = MDS2, colour = population)) +
  coord_fixed() + ## need aspect ratio of 1!
  geom_segment(data = VOC_score,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), color = VOC_score$arr_col) +
  geom_text(data = VOC_score, 
            aes(x = NMDS1, y = NMDS2, label = rowname), 
            size = 3) + 
  geom_segment(data = data.frame(x = 0.6, y = -1),
               aes(x = x, xend = (x + 0.1), y = y, yend = y),
               arrow = arrow(length = unit(0.25, "cm")), color = "grey") +
  annotate("text", x = 0.92, y = -1, label = "P > 0.1") +
  annotate("text", 
           x = 0.8, 
           y = -1.2, 
           label = paste0("stress = ", 
                          round(nmds_otus_BVOC$stress, digits = 4))) + 
  theme(legend.title = element_text(size = 13), 
        legend.text = element_blank())
show(p_fit)
#ggsave(filename = "figures/envfit_log.png", plot = p_fit)
```

caution: kaurene出してるのは"AZ_004_1", "AJ_020_1"のみ(グラフ内はAJ_020_1のみ)

###weighted UF

```{r}
pca_BVOC_log_rm_UFwei <- prcomp(data.frame(BVOC_log_UFwei[, -c(14)], 
                                     total = BVOC_totalog_UFwei), 
                          scale. = TRUE)
summary(pca_BVOC_log_rm_UFwei)
```

```{r}
set.seed(seed[7])
fit_VOC_UF <- envfit(nmds_otus_BVOC_UFwei, cbind(pca_BVOC_log_rm_UFwei$x[, 1:2], 
                                                 total = BVOC_totalog_UFwei))
show(fit_VOC_UF)

VOC_score_UF <- scores(fit_VOC_UF, display = "vectors")
VOC_score_UF <- rownames_to_column(data.frame(VOC_score_UF))
VOC_score_UF <- data.frame(VOC_score_UF, 
                        p = fit_VOC_UF$vectors$pvals, 
                        arr_col = rep("grey35", nrow(VOC_score_UF)))
```

```{r}
p_fit <- ggplot(df_nmds_BVOC_UFwei) +
  theme_bw() + 
  geom_point(mapping = aes(x = MDS1, y = MDS2, colour = population)) +
  coord_fixed(ratio = 1) + ## need aspect ratio of 1!
  geom_segment(data = VOC_score_UF,
               aes(x = 0, xend = NMDS1/10, y = 0, yend = NMDS2/10),
               arrow = arrow(length = unit(0.25, "cm")), 
               color = VOC_score_UF$arr_col, 
               alpha = 0.4) +
  geom_text(data = VOC_score_UF,  
            aes(x = NMDS1/10, y = NMDS2/10, label = rowname),
            size = 3) +
  geom_segment(data = data.frame(x = 0.035, y = -0.015),
               aes(x = x, xend = (x + 0.01), y = y, yend = y),
               arrow = arrow(length = unit(0.25, "cm")), 
               color = "grey35", 
               alpha = 0.4) +
  annotate("text", x = 0.03, y = -0.015, label = "P > 0.1") +
  annotate("text", 
           x = 0.035, y = -0.01, 
           label = paste0("stress = ", 
                          round(nmds_otus_BVOC_UFwei$stress, digits = 4))) +
  theme(legend.title = element_text(size = 13), 
        legend.text = element_blank())
p_fit
#ggsave(filename = "figures/envfit_weightedUF_log.png", plot = p_fit)
```

##envfit_logpre

```{r}
set.seed(45)
fit_logpre <- envfit(nmds_otus_BVOC, cbind(BVOC_log[,-14], total = BVOC_totalog))
fit_logpre

VOC_score_logpre <- scores(fit_logpre, display = "vectors")
VOC_score_logpre <- rownames_to_column(data.frame(VOC_score_logpre))
VOC_score_logpre <- data.frame(VOC_score_logpre, 
                        p = fit_logpre$vectors$pvals, 
                        arr_col = rep("white", nrow(VOC_score_logpre)))
VOC_score_logpre$arr_col[VOC_score_logpre$p < 0.1] <- "grey"
VOC_score_logpre$arr_col[VOC_score_logpre$p < 0.05] <- "black"

VOC_signif_logpre <- VOC_score_logpre[VOC_score_logpre$p < 0.1, ]
```

```{r, fig.asp=0.7, eval=FALSE}
plot(BVOC_log[, 13], nmds_otus_BVOC$points[, 2])
```

```{r}
p_fit_logpre <- ggplot(df_nmds_BVOC) +
  theme_bw() + 
  geom_point(mapping = aes(x = MDS1, y = MDS2, colour = population)) +
  coord_fixed() + ## need aspect ratio of 1!
  geom_segment(data = VOC_signif_logpre,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), color = VOC_signif_logpre$arr_col) +
  geom_text(data = VOC_signif_logpre, 
            aes(x = NMDS1, y = NMDS2, label = rowname), 
            size = 3) + 
  geom_segment(data = data.frame(x = 0.6, y = -1.08),
               aes(x = x, xend = (x + 0.1), y = y, yend = y),
               arrow = arrow(length = unit(0.25, "cm")), color = "black") +
  annotate("text", x = 0.9, y = -1.1, label = "P < 0.05") +
  annotate("text", 
           x = 0.8, 
           y = -1.2, 
           label = paste0("stress = ", 
                          round(nmds_otus_BVOC$stress, digits = 4)))
p_fit_logpre
#ggsave(filename = "figures/envfit_log_prePCR.png", plot = p_fit_logpre)
```

###fit abundant Class

```{r}
set.seed(seed[10])
fit_Class_BVOC <- envfit(nmds_otus_BVOC, 
                         df_Class_abundant[rownames(metadata_VOC), ])
show(fit_Class_BVOC)
```

```{r}
score <- scores(fit_Class_BVOC, display = "vectors")
score <- rownames_to_column(data.frame(score))
score <- data.frame(score, 
                    Class = tax_table(ps_rarefied_Class)[score$rowname, "Class"], 
                    p = fit_Class_BVOC$vectors$pvals,
                    arr_col = rep("grey40", nrow(score)))
score$arr_col[score$p < 0.05] <- rep("black", sum(score$p < 0.05))
```

```{r}
p_fit_logpre <- ggplot(df_nmds_BVOC) +
  theme_bw() + 
  geom_point(mapping = aes(x = MDS1, y = MDS2, colour = population)) +
  coord_fixed() + ## need aspect ratio of 1!
  geom_segment(data = VOC_signif_logpre,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), 
               color = VOC_signif_logpre$arr_col) +
  geom_text(data = VOC_signif_logpre, 
            aes(x = NMDS1, y = NMDS2, label = rowname), 
            size = 3) + 
  geom_text(data = score[score$p < 0.05, ], 
            aes(x = NMDS1, y = NMDS2, label = Class), 
            color = "grey40", 
            size = 3) + #show only significant Class
  geom_segment(data = data.frame(x = 0.6, y = -1.08),
               aes(x = x, xend = (x + 0.1), y = y, yend = y),
               arrow = arrow(length = unit(0.25, "cm")), color = "black") +
  annotate("text", x = 0.9, y = -1.1, label = "P < 0.05") +
  annotate("text", 
           x = 0.8, 
           y = -1.2, 
           label = paste0("stress = ", 
                          round(nmds_otus_BVOC$stress, digits = 4)))
p_fit_logpre
#ggsave(filename = "figures/envfit_log_Class.png", plot = p_fit_logpre)
```

###weighted UF

```{r}
set.seed(seed[8])
fit_VOC_UF <- envfit(nmds_otus_BVOC_UFwei, cbind(BVOC_log_UFwei[,-14], 
                                                 total = BVOC_totalog_UFwei))
show(fit_VOC_UF)

VOC_score_UF <- scores(fit_VOC_UF, display = "vectors")
VOC_score_UF <- rownames_to_column(data.frame(VOC_score_UF))
VOC_score_UF <- data.frame(VOC_score_UF, 
                        p = fit_VOC_UF$vectors$pvals, 
                        arr_col = rep("grey35", nrow(VOC_score_UF)))
```

```{r}
p_fit <- ggplot(df_nmds_BVOC_UFwei) +
  theme_bw() + 
  geom_point(mapping = aes(x = MDS1, y = MDS2, colour = population)) +
  coord_fixed(ratio = 1) + ## need aspect ratio of 1!
  geom_segment(data = VOC_score_UF,
               aes(x = 0, xend = NMDS1/10, y = 0, yend = NMDS2/10),
               arrow = arrow(length = unit(0.25, "cm")), 
               color = VOC_score_UF$arr_col, 
               alpha = 0.4) +
  geom_text(data = VOC_score_UF,  
            aes(x = NMDS1/10, y = NMDS2/10, label = rowname),
            size = 3) +
  geom_segment(data = data.frame(x = 0.035, y = -0.015),
               aes(x = x, xend = (x + 0.01), y = y, yend = y),
               arrow = arrow(length = unit(0.25, "cm")), 
               color = "grey35", 
               alpha = 0.4) +
  annotate("text", x = 0.03, y = -0.015, label = "P > 0.1") +
  annotate("text", 
           x = 0.035, y = -0.01, 
           label = paste0("stress = ", 
                          round(nmds_otus_BVOC_UFwei$stress, digits = 4))) +
  theme(legend.title = element_text(size = 13), 
        legend.text = element_blank())
p_fit
#ggsave(filename = "figures/envfit_weightedUF_log.png", plot = p_fit)
```
#Yet
#picrust2

```{r, eval=FALSE}
metadata_picrus <- metadata3
rownames(metadata_picrus) <- paste0(as.integer(rownames(metadata_picrus)), metadata_picrus$site)
metadata_picrus <- rownames_to_column(metadata_picrus)

abundance <-  readr::read_delim(
        paste0(path, "/Analysis/picrust2_out_ITS/pathways_out/path_abun_unstrat.tsv"),
        delim = "\t",
        escape_double = FALSE,
        trim_ws = TRUE
      )

colnames(abundance) <- c(colnames(abundance)[1], paste0(colnames(abundance)[-1], metadata_picrus$site))
```

```{r, eval=FALSE}
metacyc_daa_results_df <- pathway_daa(abundance = abundance %>% column_to_rownames("pathway"), 
                                      metadata = metadata_picrus, 
                                      group = "site", 
                                      daa_method = "DESeq2")

# Annotate MetaCyc pathway results without KO to KEGG conversion
metacyc_daa_annotated_results_df <- pathway_annotation(pathway = "MetaCyc", 
                                                       daa_results_df = metacyc_daa_results_df, 
                                                       ko_to_kegg = FALSE)

# Generate pathway error bar plot
# Please change column_to_rownames() to the feature column
# Please change Group to metadata$your_group_column if you are not using example dataset
#bar_picrus <- pathway_errorbar(abundance = abundance %>% column_to_rownames("pathway"), daa_results_df = metacyc_daa_annotated_results_df, Group = metadata_picrus$site, ko_to_kegg = FALSE, p_values_threshold = 0.05, order = "group", select = NULL, p_value_bar = TRUE, colors = NULL, x_lab = "description")

# Generate pathway heatmap
# Please change column_to_rownames() to the feature column if you are not using example dataset
# Please change group to "your_group_column" if you are not using example dataset
feature_with_p_0.05 <- metacyc_daa_annotated_results_df %>% filter(p_adjust < 0.05)
heat_picrus <- pathway_heatmap(abundance = abundance %>% filter(pathway %in% feature_with_p_0.05$feature) %>% column_to_rownames("pathway"), 
                               metadata = metadata_picrus, 
                               group = "site")

show(heat_picrus)
```
